static/app/components/dropdownMenuControl.tsx

```diff
@@ -141,45 +142,40 @@ function DropdownMenuControl({
   position = 'bottom-start',
   isDismissable = true,
   shouldCloseOnBlur = true,
+  shouldCloseOnInteractOutside,
+  preventOverflowOptions,
   ...props
 }: DropdownMenuControlProps) {
   const isDisabled = disabledProp ?? (!items || items.length === 0);
 
+  const {rootOverlayState} = useContext(DropdownMenuContext);
   const {
     isOpen,
-    state,
+    state: overlayState,
     triggerRef,
     triggerProps: overlayTriggerProps,
     overlayProps,
   } = useOverlay({
-    onClose: closeRootMenu,
     isOpen: isOpenProp,
+    onClose: rootOverlayState?.close,
     offset,
     position,
-    isDismissable: !isSubmenu && isDismissable,
-    shouldCloseOnBlur: !isSubmenu && shouldCloseOnBlur,
-    shouldCloseOnInteractOutside: () => !isSubmenu,
-    // Necessary for submenus to be correctly positioned
-    ...(isSubmenu && {preventOverflowOptions: {boundary: document.body, altAxis: true}}),
+    isDismissable,
+    shouldCloseOnBlur,
+    shouldCloseOnInteractOutside,
+    preventOverflowOptions,
   });
 
   const {menuTriggerProps, menuProps} = useMenuTrigger(
     {type: 'menu', isDisabled},
-    {...state, focusStrategy: 'first'},
+    {...overlayState, focusStrategy: 'first'},
     triggerRef
   );
 
   const {buttonProps} = useButton(
     {
       isDisabled,
       ...menuTriggerProps,
-      ...(isSubmenu && {
-        onKeyUp: e => e.continuePropagation(),
-        onKeyDown: e => e.continuePropagation(),
-        onPress: () => null,
-        onPressStart: () => null,
-        onPressEnd: () => null,
-      }),
```

Comment:
Moved to `DropdownMenu`
https://github.com/getsentry/sentry/pull/43581/files#diff-8381db94caec14ffb1be1724f14db1f1afeb916cb1e591244915a7d6116d8540R161-R169